### PR TITLE
[#25] Improved UI of Event cards and set up Google Sheets backend (BLOCKED)

### LINKED ISSUE #25: Beautify Event Cards and Connect with Google Sheets "Database"
- Improve UI for events on bottom of landing page
- Should also be able to work for 'no events'
- Connect to google sheets for easy write/read access with all founders

### PR DESCRIPTION
**Pull request by rohankumar2025**
Screenshot 2024-09-10 at 4 07 15 PM

- Set up better UI for event cards (some changes will likely still be needed as it isn't perfect)
- Set up backend using Google Sheets (removed hardcoded events from codebase)
- Added enum for BranchType to improve readability and so that it could be stored in database well
- Stored client secrets / api keys as environment variables (see Google Drive for directions on how to use)

- Resolves #25

### CODE REVIEW

**Comment by smmzhu** [comment]
Waiting for rebasing with main for vercel ci run. Ping me to verify once done

**Review by dayvidwang — APPROVED** [review]
Nice work! Very clean. 

Few minor comments but nothing blocking, Sam should probably take a look as well.

For the google sheet itself, it may be a good idea to connect a google form to it so that there's input validation when officers enter new events.
e.g. datetime
image

e.g. regex validation
image

File: src/apis/getSheetsData.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+enum SheetName {
+  Events = "Events",
+}
+
+const getRawSheetData = async (sheetName: SheetName) => {
+  const auth = new google.auth.GoogleAuth({
+    credentials: {
+      client_email: process.env.GOOGLE_SERVICE_ACCOUNT_EMAIL,
+      private_key: process.env.GOOGLE_PRIVATE_KEY?.replace(/\\n/g, "\n"),
```
[the remark this answers is not in the document]
Comment (rohankumar2025) [reply src/apis/getSheetsData.ts:12]: Yes this is basically taken from the documentation I found

**Comment by rohankumar2025** [comment]
> Nice work! Very clean.
> 
> Few minor comments but nothing blocking, Sam should probably take a look as well.
> 
> For the google sheet itself, it may be a good idea to connect a google form to it so that there's input validation when officers enter new events. e.g. datetime image
> 
> e.g. regex validation image

I like this idea a lot We should def do this

**Comment by rohankumar2025** [comment]
Blocked by vercel deployment ci (need to be able to inject secrets into deployment)

**Pull request merged**